CMakeLists.txt

```diff
@@ -1159,6 +1159,21 @@ if(CONFIG_BUILD_OUTPUT_BIN)
   endif()
 endif()
 
+if(CONFIG_BUILD_OUTPUT_BIN AND CONFIG_BUILD_OUTPUT_UF2)
```

Comment:
I think UF2 deserves to be a Zephyr module of its own, and therefore this code should go into the CMakeLists.txt code for that module.

This will also ensure the options are only available when the module is available.

Reply:
Making sure I understand this correctly... When #30904 is merged, the only thing in the actual external module needed will be the `zephyr/module.yml`, and then the `CMakeLists.txt` and `Kconfig.uf2` files would be in the main Zephyr repo, and only included if that uf2 module was also fetched by west and was present?

Reply:
that is correctly understood.